test/suites/osconfig/systemd-resolved.robot

```diff
@@ -0,0 +1,167 @@
+Resource            ../../resources/systemd.resource
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+
+*** Variables ***
+${KUBELET_CONFIG_FILE}      /var/lib/microshift/resources/kubelet/config/config.yaml
+${RESOLVE_CONF_FILE}        /run/systemd/resolve/resolv.conf
+
+
+*** Test Cases ***
+Verify Kubelet Config With Systemd-Resolved Running
+    [Documentation]    Verify kubelet uses the upstream resolve file of
+    ...    systemd-resolved when the service is running
+    [Setup]    Run Keywords
+    ...    Systemd-Resolved Must Be Installed And Disabled
+    ...    Start Systemd-Resolved
+
+    Delete Kubelet Configuration File
+    Restart MicroShift
+
+    # Verify the presence of the kubelet option
+    ${rc}=    Execute Command
+    ...    grep -q "resolvConf: ${RESOLVE_CONF_FILE}" ${KUBELET_CONFIG_FILE}
+    ...    sudo=True    return_rc=True    return_stdout=False    return_stderr=False
+    Should Be Equal As Integers    0    ${rc}
+
+    [Teardown]    Run Keywords
+    ...    Stop Systemd-Resolved
+
+Verify Kubelet Config With Systemd-Resolved Disabled
+    [Documentation]    Verify kubelet does not use the upstream resolve file of
+    ...    systemd-resolved when the service is disabled
+    [Setup]    Run Keywords
+    ...    Systemd-Resolved Must Be Installed And Disabled
+
+    Delete Kubelet Configuration File
+    Restart MicroShift
+
+    # Verify the absence of the kubelet option
+    ${rc}=    Execute Command
+    ...    grep -q "resolvConf: ${RESOLVE_CONF_FILE}" ${KUBELET_CONFIG_FILE}
+    ...    sudo=True    return_rc=True    return_stdout=False    return_stderr=False
+    Should Not Be Equal As Integers    0    ${rc}
+
+Verify Kubelet Config With Systemd-Resolved Uninstalled
+    [Documentation]    Verify kubelet does not use the upstream resolve file of
+    ...    systemd-resolved when the package is not present
+    [Setup]    Run Keywords
+    ...    Uninstall Systemd-Resolved
+    ...    Systemd-Resolved Must Not Be Installed
+
+    Delete Kubelet Configuration File
+    Restart MicroShift
+
+    # Verify the absence of the kubelet option
+    ${rc}=    Execute Command
+    ...    grep -q "resolvConf: ${RESOLVE_CONF_FILE}" ${KUBELET_CONFIG_FILE}
+    ...    sudo=True    return_rc=True    return_stdout=False    return_stderr=False
+    Should Not Be Equal As Integers    0    ${rc}
+
+    # Revert the system to the original configuration
+    [Teardown]    Run Keywords
+    ...    Restore Systemd-Resolved
+    ...    Systemd-Resolved Must Be Installed And Disabled
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Systemd-Resolved Must Be Installed And Disabled
+    [Documentation]    Verify the systemd-resolved package is installed
+
+    ${stdout}    ${stderr}    ${rc}=    Execute Command
```

Comment:
This step will fail if the tests are run out of order, right? I don't see anything installing the package if it is missing.

Reply:
Could we just run this test on an RPM-based system where we can add and remove packages easily?

Reply:
It's a keyword, so when it is called, the environment should be appropriate both for ostree and rpm systems. 
I do not think we need to fall back to rpm-only in this case.

Reply:
The assumption is that the package is installed because of this change: 
https://github.com/openshift/microshift/pull/2441/files/52e579fd21ce61369ba14bfbd42c9383fdd59354#diff-fa38d74133cee0980c67ee278c2e94c103b03d54fff633f4825b978c7d88707aR29

Reply:
The tests run in a random order, though, and some of the other tests remove the package.

Reply:
OK, I missed line 72.